### ISSUE TITLE
Edited tracks scrobbling unedited

### ISSUE THREAD

**Issue by EvertonRoberto** [web-scrobbler/web-scrobbler#2403]
I have almost 800 edited tracks and recently it started scrobbling unedited, looks like the tracks never got edited and I have to re-edit. Once I re-edit it, it works fine. There's a way to fix this? Please tell me I don't have to re-edit all the tracks.

![6846845](https://user-images.githubusercontent.com/65428796/82109404-e889c980-970b-11ea-84c9-1a172d82aeb4.png)

<details>
  <summary>Debug log</summary>
  Here is the content of debug log:

  ```
  ﻿
wrapper.js:69 storage.Options = {
  "disableGa": false,
  "disabledConnectors": "[hidden]",
  "forceRecognize": false,
  "scrobblePercent": 50,
  "scrobblePodcasts": false,
  "useNotifications": true,
  "useUnrecognizedSongNotifications": true
}
wrapper.js:69 storage.ListenBrainz = {}
wrapper.js:69 storage.Connectors = {
  "Tidal": {
    "useShortTrackNames": false
  },
  "YouTube": {
    "scrobbleEntertainmentOnly": false,
    "scrobbleMusicOnly": false
  }
}
wrapper.js:69 storage.LastFM = {
  "sessionID": "********************************",
  "sessionName": "***************"
}
wrapper.js:69 storage.LibreFM = {}
wrapper.js:69 storage.Core = {
  "appVersion": "2.31.0"
}
scrobble-service.js:63 Bind Last.fm scrobbler
2
scrobble-service.js:49 Unable to bind Libre.fm
2
scrobble-service.js:49 Unable to bind ListenBrainz
inject.js:52 Injecting vendor/filter.js
inject.js:52 Injecting vendor/jquery.min.js
inject.js:52 Injecting vendor/browser-polyfill.min.js
inject.js:52 Injecting core/content/util.js
inject.js:52 Injecting core/content/reactor.js
inject.js:52 Injecting core/content/connector.js
inject.js:52 Injecting connectors/spotify.js
inject.js:52 Injecting core/content/starter.js
util.js:40 Tab 3: Created controller for Spotify connector
util.js:40 Tab 3: New song detected: {
  "track": "The girl next door",
  "artist": "NU'EST",
  "album": "THE SECOND MINI ALBUM Hello",
  "albumArtist": null,
  "uniqueID": null,
  "duration": 217,
  "currentTime": 23,
  "isPlaying": true,
  "trackArt": "https://i.scdn.co/image/ab67616…",
  "isPodcast": false
}
scrobble-service.js:91 Send "get info" request: 1
util.js:40 Last.fm: track.getinfo response:
{
  "track": {
    "name": "The Girl Next Door",
    "url": "https://www.last.fm/music/NU%27EST/_/The+Girl+Next+Door",
    "duration": "217000",
    "streamable": {
      "#text": "0",
      "fulltrack": "0"
    },
    "listeners": "820",
    "playcount": "3913",
    "artist": {
      "name": "NU'EST",
      "mbid": "8c00b1aa-68f6-4874-98cb-de24c25ffe67",
      "url": "https://www.last.fm/music/NU%27EST"
    },
    "album": {
      "artist": "NU'EST",
      "title": "THE SECOND MINI ALBUM HELLO",
      "url": "https://www.last.fm/music/NU%27EST/THE+SECOND+MINI+ALBUM+HELLO",
      "image": [
        {
          "#text": "https://lastfm.freetls.fastly.net/i/u/34s/490ed8d….png",
          "size": "small"
        },
        {
          "#text": "https://lastfm.freetls.fastly.net/i/u/64s/490ed8d….png",
          "size": "medium"
        },
        {
          "#text": "https://lastfm.freetls.fastly.net/i/u/174s/490ed8d….png",
          "size": "large"
        },
        {
          "#text": "https://lastfm.freetls.fastly.net/i/u/300x300/490ed8d….png",
          "size": "extralarge"
        }
      ]
    },
    "userplaycount": "0",
    "userloved": "0",
    "toptags": {
      "tag": []
    }
  }
}
coverartarchive.js:10 Using local/parsed artwork
util.js:40 Tab 3: Song finished processing: {
  "parsed": {
    "track": "The girl next door",
    "artist": "NU'EST",
    "album": "THE SECOND MINI ALBUM Hello",
    "albumArtist": null,
    "uniqueID": null,
    "duration": 217,
    "currentTime": 23,
    "isPlaying": true,
    "trackArt": "https://i.scdn.co/image/ab67616…",
    "isPodcast": false
  },
  "processed": {
    "track": "The Girl Next Door",
    "album": null,
    "artist": "NU'EST",
    "albumArtist": null,
    "duration": 217
  },
  "flags": {
    "isScrobbled": false,
    "isCorrectedByUser": false,
    "isValid": true,
    "isMarkedAsPlaying": false,
    "isSkipped": false,
    "isReplaying": false
  },
  "metadata": {
    "userloved": false,
    "startTimestamp": 1589599250,
    "label": "Spotify",
    "trackArtUrl": "https://lastfm.freetls.fastly.net/i/u/300x300/490ed8d….png",
    "artistUrl": "https://www.last.fm/music/NU%27EST",
    "trackUrl": "https://www.last.fm/music/NU%27EST/_/The+Girl+Next+Door",
    "albumUrl": "https://www.last.fm/music/NU%27EST/THE+SECOND+MINI+ALBUM+HELLO",
    "userPlayCount": 0
  },
  "internalId": "1593745e617acb31b1837d438c759444",
  "connectorLabel": "Spotify"
}
util.js:40 Tab 3: The song will be scrobbled in 108 seconds
util.js:40 Tab 3: The song will be repeated in 217 seconds
scrobble-service.js:107 Send "now playing" request: 1
util.js:40 Last.fm: track.updatenowplaying response:
{
  "nowplaying": {
    "artist": {
      "corrected": "0",
      "#text": "NU'EST"
    },
    "ignoredMessage": {
      "code": "0",
      "#text": ""
    },
    "album": {
      "corrected": "0",
      "#text": "THE SECOND MINI ALBUM Hello"
    },
    "albumArtist": {
      "corrected": "0",
      "#text": ""
    },
    "track": {
      "corrected": "0",
      "#text": "The Girl Next Door"
    }
  }
}
util.js:40 Tab 3: Song set as now playing
O DevTools não carregou o SourceMap: Não foi possível carregar o conteúdo para chrome-extension://hhinaapppaileiechjoiifaancjggfjm/vendor/browser-polyfill.min.js.map: Erro de HTTP: código de status 404, net::ERR_UNKNOWN_URL_SCHEME
O DevTools não carregou o SourceMap: Não foi possível carregar o conteúdo para chrome-extension://hhinaapppaileiechjoiifaancjggfjm/vendor/md5.min.js.map: Erro de HTTP: código de status 404, net::ERR_UNKNOWN_URL_SCHEME
scrobble-service.js:123 Send "scrobble" request: 1
util.js:40 Last.fm: track.scrobble response:
{
  "scrobbles": {
    "@attr": {
      "accepted": 1,
      "ignored": 0
    },
    "scrobble": {
      "artist": {
        "corrected": "0",
        "#text": "NU'EST"
      },
      "ignoredMessage": {
        "code": "0",
        "#text": ""
      },
      "albumArtist": {
        "corrected": "0",
        "#text": ""
      },
      "timestamp": "1589599250",
      "album": {
        "corrected": "0",
        "#text": "THE SECOND MINI ALBUM Hello"
      },
      "track": {
        "corrected": "0",
        "#text": "The Girl Next Door"
      }
    }
  }
}
util.js:40 Tab 3: Scrobbled successfully
﻿
​

  ```
</details>

English is no my first language, sorry for any mistake. And thank you for your great work.

**Comment by alexesprit** [comment]
@EvertonRoberto Hi!

**TL; DR**: This was broken due to a new update where an album info was added to the Spotify connector. I have an idea how to fix the issue, but I need to implement it and test firstly.

> I have almost 800 edited tracks and recently it started scrobbling unedited, looks like the tracks never got edited and I have to re-edit. Once I re-edit it, it works fine

Edited tracks are stored and accessed by a unique ID (a freeform string). Unfortunately, Spotify connector doesn't provide this unuque ID from Spotify, so the extension generates this ID with using an artist name, a track title, and an album title.

Previously, Spotify connector didn't provide an album name, and the extension generated unique IDs with an artist name and a track title only. I changed this in the latest version (v2.31.0). So, unique IDs becames invalid in v2.31.0, as now they calculated differently.

I didn't realize this is an issue, although such breaking changes were made in other connectors (less popular, though).

> There's a way to fix this?

The only way to fix the issue is rewrite the extension somehow to get the "old" saved data correctly. I think I have an idea how to do that.

**Comment by alexesprit** [comment]
@EvertonRoberto  Should be fixed in v2.31.1. Please verify it's resolved for you. :)

**Comment by EvertonRoberto** [comment]
> @EvertonRoberto Should be fixed in v2.31.1. Please confirm it's resolved for you. :)

Ok. Thank you ;)
